UnityProject/Assets/Scripts/HealthV2/Living/Metabolism/DigestiveSystemBase.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections.Generic;
+using System;
+using Chemistry;
+using UnityEngine;
+using Player.Movement;
+using Items.Implants.Organs;
+
+namespace HealthV2
+{
+	public class DigestiveSystemBase : MonoBehaviour, IMovementEffect
+	{
+		//Assuming 20 bodyparts (including organs) on average with a consumption of one.
+		//That is 2 Hunger per second. We want around 0.055 per second to have a full stomach (100u) last half an hour
+		//2 / 0.055 is 36.
+		private const float HUNGER_DEPLETION_DIVIDER = 36f;
+		private float hungerConsumptionRate = 20;
+
+		private bool isActive = true;
+		private float secondsToProcess = 10f;
+		public float CurrentHunger { get; private set; } = 0;
+		private float maxHunger = 100;
+
+		private List<IStomachProcess> stomachList = new List<IStomachProcess>();
+		public List<BodyFat> BodyFat { get; private set; } = new List<BodyFat>();
+
+		private LivingHealthMasterBase livingHealthMaster;
+
+		public HungerState HungerState
+		{
+			get
+			{
+				return GetCurrentHungerState();
+			}
+		}
+
+		#region MovementFields
+
+		[SerializeField] private float maxRunSpeedDebuff = -2;
+		[SerializeField] private float maxWalkingDebuff = -1.5f;
+		[SerializeField] private float maxCrawlDebuff = -0.2f;
+
+		public float RunningSpeedModifier { get; private set; }
+
+		public float WalkingSpeedModifier { get; private set; }
+
+		public float CrawlingSpeedModifier { get; private set; }
+
+		#endregion
+
+		public void OnEnable()
+		{
+			if (isActive) UpdateManager.Add(ProcessStomachs, secondsToProcess);
+		}
+
+		public void OnDisable()
+		{
+			if (isActive) UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, ProcessStomachs);
+		}
+
+		private void ProcessStomachs()
+		{
+			hungerConsumptionRate = 0;
+			foreach (BodyPart part in livingHealthMaster.BodyPartList) //Consume food from body parts
+			{
+				hungerConsumptionRate += part.HungerConsumption / HUNGER_DEPLETION_DIVIDER;
+			}
+			CurrentHunger = Math.Max(0, CurrentHunger - hungerConsumptionRate); 
+
+
+			List<IStomachProcess> stomachs = new List<IStomachProcess>(stomachList);
+			foreach (IStomachProcess stomach in stomachs) //Get Stomachs to digest food
+			{
+				if (stomach == null)
+				{
+					stomachList.Remove(stomach);
+					continue;
+				}
+				if (CurrentHunger >= maxHunger) break;
+				CurrentHunger += stomach.ProcessContent();
```

Comment:
I think it would better not to interact with stomachs directly, 
treat the blood pool as another int , and don't touch stomachs

Reply:
might have to exlain what you mean by this, i dont quite undertand

Reply:
so, stomachs just released into the bloodstream, 
if there is enough nutriment in the bloodstream digestive system, just takes from the fat that is available, 
and then when the fat runs out its Starts taking from the bloodstream from the bloodstream
if there's too much nutriment in the bloodstream, it will put it into the fats

so if you take a sample of someone's blood there will be nutriment in it, and also means you can directly inject nutriment into someone,  also less hardcoded connections as well

Reply:
So in this proposal, Digestive system will then just sample the blood stream to get hunger?

Reply:
to get hunger you can still do the total amount of nutriment available
bloodstream[nutriment] + fats. totalstorage

Reply:
The issue with this, is that it doesnt support different nutrients on a per stomach basis. Like a moth stomach can digest cloth for food but a human stomach cannot. It means you cant inject someone with nutriment yes, but from my knowledge (though I could be wrong) you cant just grab a syringe of carbohydrates and inject it into someone.